src/main/webapp/app/entities/study/study.pipe.ts

```diff
@@ -0,0 +1,32 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Participant} from '../participant/participant.model'
+
+@Pipe({
+    name: 'participant',
+    pure: false
+})
+export class StudyPipe implements PipeTransform {
+    transform(participants: Participant[], filter: Participant): Participant[] {
+        if (!participants || !filter) {
+            return participants;
+        }
+        return participants.filter((participant: Participant) => this.applyFilter(participant, filter));
+    }
+
+    applyFilter(participant: Participant, filter: Participant): boolean {
+        for (const field in filter) {
+            if (filter[field]) {
+                if (typeof filter[field] === 'string') {
+                    if (participant[field].toLowerCase().indexOf(filter[field].toLowerCase()) === -1) {
+                        return false;
```

Comment:
If you try to input text into a filter that the participants don't have (i.e Location), this throws an error and you can't save the Study